### PR TITLE
Remove old, unused HTML bits from the Makefile

### PR DESCRIPTION
**Pull request by wilzbach**
Also removes the redundant specification of the Ddoc files to dlang.org - this repository
was already required to be exist before.

### CODE REVIEW

File: posix.mak [hunk 1]
```diff
@@ -76,6 +76,7 @@ endif
 # Configurable stuff that's rarely edited
 INSTALL_DIR = ../install
 DRUNTIME_PATH = ../druntime
+DLANG_ORG_DIR = ../dlang.org
```
Comment (wilzbach) [inline posix.mak]: Maybe at some point in the future we get https://github.com/dlang/dmd/pull/7476 ...

File: posix.mak [hunk 2]
```diff
@@ -84,19 +85,6 @@ TOOLS_DIR=../tools
...
 DSCANNER_DIR=../dscanner-$(DSCANNER_HASH)
 
-# Documentation-related stuff
-DOCSRC = ../dlang.org
-WEBSITE_DIR = ../web
-DOC_OUTPUT_DIR = $(WEBSITE_DIR)/phobos-prerelease
-BIGDOC_OUTPUT_DIR = /tmp
-SRC_DOCUMENTABLES = index.d $(addsuffix .d,$(STD_MODULES) \
-	$(EXTRA_DOCUMENTABLES))
-STDDOC = $(DOCSRC)/html.ddoc $(DOCSRC)/dlang.org.ddoc $(DOCSRC)/std_navbar-prerelease.ddoc $(DOCSRC)/std.ddoc $(DOCSRC)/macros.ddoc $(DOCSRC)/.generated/modlist-prerelease.ddoc
```
Comment (wilzbach) [inline posix.mak]: This is overwritten by dlang.org

File: posix.mak [hunk 3]
```diff
@@ -84,19 +85,6 @@ TOOLS_DIR=../tools
 DSCANNER_HASH=bb32e9f1e3e5206deb11b3dbc43ad44e23fabf96
 DSCANNER_DIR=../dscanner-$(DSCANNER_HASH)
 
-# Documentation-related stuff
-DOCSRC = ../dlang.org
-WEBSITE_DIR = ../web
-DOC_OUTPUT_DIR = $(WEBSITE_DIR)/phobos-prerelease
```
Comment (wilzbach) [inline posix.mak]: Overwritten by dlang.org

File: posix.mak [hunk 4]
```diff
@@ -497,24 +490,11 @@ $(foreach p,$(SRC_DOCUMENTABLES),$(eval \
...
-rsync-prerelease : html
-	rsync -avz $(DOC_OUTPUT_DIR)/ d-programming@digitalmars.com:data/phobos-prerelease/
-	rsync -avz $(WEBSITE_DIR)/ d-programming@digitalmars.com:data/phobos-prerelase/
-
-html_consolidated :
-	$(DDOC) -Df$(DOCSRC)/std_consolidated_header.html $(DOCSRC)/std_consolidated_header.dd
-	$(DDOC) -Df$(DOCSRC)/std_consolidated_footer.html $(DOCSRC)/std_consolidated_footer.dd
-	$(MAKE) -f $(MAKEFILE) DOC_OUTPUT_DIR=$(BIGDOC_OUTPUT_DIR) STDDOC=$(BIGSTDDOC) html -j 8
-	cat $(DOCSRC)/std_consolidated_header.html $(BIGHTMLS)	\
-	$(DOCSRC)/std_consolidated_footer.html > $(DOC_OUTPUT_DIR)/std_consolidated.html
```
Comment (wilzbach) [inline posix.mak]: AFAICT this isn't used anymore

File: posix.mak [hunk 5]
```diff
@@ -497,24 +490,11 @@ $(foreach p,$(SRC_DOCUMENTABLES),$(eval \
...
   $(DDOC) project.ddoc $(STDDOC) -Df$$@ $$<))
 
-html : $(DOC_OUTPUT_DIR)/. $(HTMLS) $(STYLECSS_TGT)
-
-allmod :
-	@echo $(SRC_DOCUMENTABLES)
-
-rsync-prerelease : html
-	rsync -avz $(DOC_OUTPUT_DIR)/ d-programming@digitalmars.com:data/phobos-prerelease/
-	rsync -avz $(WEBSITE_DIR)/ d-programming@digitalmars.com:data/phobos-prerelase/
```
Comment (wilzbach) [inline posix.mak]: This is handled by dlang.org and DAutoTest now. There's still an unused `rsync` deployment target at dlang.org.

File: posix.mak [hunk 6]
```diff
@@ -497,24 +490,11 @@ $(foreach p,$(SRC_DOCUMENTABLES),$(eval \
 $(DOC_OUTPUT_DIR)/$(call D2HTML,$p) : $p $(STDDOC) ;\
   $(DDOC) project.ddoc $(STDDOC) -Df$$@ $$<))
 
-html : $(DOC_OUTPUT_DIR)/. $(HTMLS) $(STYLECSS_TGT)
-
-allmod :
-	@echo $(SRC_DOCUMENTABLES)
```
Comment (wilzbach) [inline posix.mak]: This was most likely added for debugging. It's never used nor referenced.

File: posix.mak [hunk 7]
```diff
@@ -497,24 +490,11 @@ $(foreach p,$(SRC_DOCUMENTABLES),$(eval \
...
-	$(DDOC) -Df$(DOCSRC)/std_consolidated_footer.html $(DOCSRC)/std_consolidated_footer.dd
-	$(MAKE) -f $(MAKEFILE) DOC_OUTPUT_DIR=$(BIGDOC_OUTPUT_DIR) STDDOC=$(BIGSTDDOC) html -j 8
-	cat $(DOCSRC)/std_consolidated_header.html $(BIGHTMLS)	\
-	$(DOCSRC)/std_consolidated_footer.html > $(DOC_OUTPUT_DIR)/std_consolidated.html
+# this target is called by dlang.org
+html : $(DOC_OUTPUT_DIR)/. $(HTMLS)
 
-changelog.html: changelog.dd
-	$(DMD) -Df$@ $<
+build-html:
```
Comment (wilzbach) [inline posix.mak]: I'm not too happy about this. I would prefer to use `html`, but then we would have to change `dlang.org` to softly detect if a new, special target `html-build` target is existent and wait for the next stable release before we can use just `html` here and then remove the soft fallback at dlang.org afterwards.
Though if wanted we could still do that - it's mostly unrelated to this PR though.

File: posix.mak [hunk 8]
```diff
@@ -497,24 +490,11 @@ $(foreach p,$(SRC_DOCUMENTABLES),$(eval \
...
-html_consolidated :
-	$(DDOC) -Df$(DOCSRC)/std_consolidated_header.html $(DOCSRC)/std_consolidated_header.dd
-	$(DDOC) -Df$(DOCSRC)/std_consolidated_footer.html $(DOCSRC)/std_consolidated_footer.dd
-	$(MAKE) -f $(MAKEFILE) DOC_OUTPUT_DIR=$(BIGDOC_OUTPUT_DIR) STDDOC=$(BIGSTDDOC) html -j 8
-	cat $(DOCSRC)/std_consolidated_header.html $(BIGHTMLS)	\
-	$(DOCSRC)/std_consolidated_footer.html > $(DOC_OUTPUT_DIR)/std_consolidated.html
+# this target is called by dlang.org
+html : $(DOC_OUTPUT_DIR)/. $(HTMLS)
 
-changelog.html: changelog.dd
```
Comment (wilzbach) [inline posix.mak]: We use a changelog folder nowadays. `changelog.dd` has been removed more than a year ago.

**Comment by quickfur** [comment]
This has been rotting here for far too long.  Is it safe to pull the trigger, even though I'm not 100% well-versed with the intricacies of the Phobos makefiles?

File: posix.mak (hunk 7 above)
Comment (quickfur) [reply posix.mak]: My only fear here is if people building Phobos from source have been relying on `make -f posix.mak html` to generate documentation. (I have been. :-P) In any case, if/once we merge this, we should update the wiki Phobos build instructions accordingly.

Comment (wilzbach) [reply posix.mak]: Hmm I don't think the wiki mentions this anywhere. I searched on these pages:

https://wiki.dlang.org/Building_under_Posix
https://wiki.dlang.org/Contributing_to_Phobos
https://wiki.dlang.org/Starting_as_a_Contributor
https://wiki.dlang.org/Contributing_to_dlang.org

**Comment by wilzbach** [comment]
> Is it safe to pull the trigger,

Yeah we have added more and more CIs to test that.
The things which aren't tested are:
- nightly release build (it's run only on dlang.org). However the release builder uses `dlang.org` correctly:

https://github.com/dlang/installer/blob/master/create_dmd_release/create_dmd_release.d#L514

- distro releases

I had a quick look around:
- ArchLinux just downloads the HTML from the releases archives (https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/dmd)
- the docker image dmd.snap comes without documentation

File: posix.mak (hunk 7 above)
Comment (quickfur) [reply posix.mak]: OK, I may have remembered wrong. In that case, merge ahoy!

**Comment by quickfur** [comment]
Alright, then, let's shoot and clean up the mess afterwards. :-D

**Pull request merged**